types/time_test.go

```diff
@@ -115,6 +115,13 @@ func (s *testTimeSuite) TestDateTime(c *C) {
 		{"2020-05-28 23:59:59T T00:00:00", "2020-05-28 23:59:59"},
 		{"2020-10-22 10:31-10:12", "2020-10-22 10:31:10"},
 		{"2018.01.01 01:00:00", "2018-01-01 01:00:00"},
+
+		// For issue 26151
```

Comment:
may we add the original case `select year('2020|5|4|||');`

Reply:
OK, i will add more test case including the original case